src/algorithms/superviewport.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AbstractViewportAlgorithm,
+  AlgorithmInput,
+  AlgorithmOutput,
+} from "./core";
+import SuperCluster, { ClusterFeature } from "supercluster";
+import { MarkerUtils, Marker } from "../marker-utils";
+import { Cluster } from "../cluster";
+import { getPaddedViewport } from "./utils";
+import equal from "fast-deep-equal";
+
+export type SuperClusterViewportOptions = SuperCluster.Options<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  { [name: string]: any },
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  { [name: string]: any }
+> & { viewportPadding?: number };
+
+export interface SuperClusterViewportState {
+  zoom: number;
+  view: [number, number, number, number];
+}
+
+/**
+ * A very fast JavaScript algorithm for geospatial point clustering using KD trees.
+ *
+ * @see https://www.npmjs.com/package/supercluster for more information on options.
+ */
+export class SuperClusterViewportAlgorithm extends AbstractViewportAlgorithm {
+  protected superCluster: SuperCluster;
+  protected markers: Marker[];
+  protected clusters: Cluster[];
+  protected state: SuperClusterViewportState;
+
+  constructor({
+    maxZoom,
+    radius = 60,
+    viewportPadding = 60,
+    ...options
+  }: SuperClusterViewportOptions) {
+    super({ maxZoom, viewportPadding });
+
+    this.superCluster = new SuperCluster({
+      maxZoom: this.maxZoom,
+      radius,
+      ...options,
+    });
+
+    this.state = { zoom: 0, view: [0, 0, 0, 0] };
```

Comment:
suggest `zoom: -1` (so that whatever the zoom the first time is always a change)

Reply:
Done